### PR TITLE
Add filterable search-able meta keys 

### PR DESCRIPTION
**Pull request by turtlepod**
Currently WPJM search all meta value,
This will cause issue explained in https://github.com/Automattic/WP-Job-Manager/issues/900

> When searched for keywords wp-content, tent, wp, content, uploads,etc all listings are displayed

### CODE REVIEW

File: wp-job-manager-functions.php [hunk 1]
```diff
@@ -177,9 +177,38 @@ function get_job_listings( $args = array() ) {
...
+		/* Search Post Title  */
 		$conditions[] = "{$wpdb->posts}.post_title LIKE '%" . esc_sql( $job_manager_keyword ) . "%'";
-		$conditions[] = "{$wpdb->posts}.ID IN ( SELECT post_id FROM {$wpdb->postmeta} WHERE meta_value LIKE '%" . esc_sql( $job_manager_keyword ) . "%' )";
+
+		/* Search Post Meta */
+		if( apply_filters( 'job_listing_search_post_meta', true ) ){
+
+			/* Only selected meta keys */
+			if( $searchable_meta_keys ){
+				$conditions[] = "{$wpdb->posts}.ID IN ( SELECT post_id FROM {$wpdb->postmeta} WHERE meta_key IN ( '" . implode( "','", $searchable_meta_keys ) . "' ) AND meta_value LIKE '%" . esc_sql( $job_manager_keyword ) . "%' )";
```
Comment (jom) [inline wp-job-manager-functions.php]: Results from the new filter must be escaped before putting it in a SQL statement.

Like: `implode( "','", array_map( 'esc_sql', $searchable_meta_keys ) )`

**Review by jom — CHANGES_REQUESTED** [review]
Thanks for this PR! This will be a big improvement.

I've added one change request. In addition, we're going to start enforcing WordPress coding and documentation standards. We'll be cleaning up core slowly, but new code should conform. Only wee things I can see are single line comments should use `//` format (https://make.wordpress.org/core/handbook/best-practices/inline-documentation-standards/php/#5-1-single-line-comments) and spaces around `if (...) {}` (https://make.wordpress.org/core/handbook/best-practices/coding-standards/php/#space-usage).

**Comment by jom** [comment]
Tested. Looks good!

**Comment by tripflex** [comment]
This is going to cause a huge issue with searching custom fields that users have created manually via PHP or with my plugin ... i'll have to figure out how to handle this, maybe just have to add all custom fields via filter to make it work.

... but those who added their own custom fields via PHP, will have issues when this is released into the wild.

**Comment by tripflex** [comment]
Also going to cause issues for themes or other plugins that have added custom fields they want searchable through keywords, which will stop working after this is released.

**Comment by spencerfinnell** [comment]
I'm not sure people really realized meta was being searched to begin with. WordPress doesn't look in meta with its built in search. 

@tripflex You could filter it like you said or you could disable it completely with the `job_listing_searchable_meta_keys` filter.

**Comment by tripflex** [comment]
@spencerfinnell yeah i'll probably just disable it by returning **false** in the `job_listing_searchable_meta_keys` filter, and then handling the actual issue as you mentioned in #900 to blacklist certain meta values, or something ... hmm not sure yet, but will have to figure something out as I know this will cause tons of support requests about keywords not working anymore.

**Comment by tripflex** [comment]
Major concern is going to be other themes or plugins that specifically expect all meta to be searched through keywords, so it would be best @kraftbj to include something in changelog that specifically states that ONLY those meta keys will be searched through keywords if the user updates

**Comment by spencerfinnell** [comment]
Are there other themes that rely on that you think? I've never really thought about it/expected it with Listify. But you certainly have better insight on expectations/how people use custom fields...

**Comment by jom** [comment]
They should be able to hook into `job_listing_searchable_meta_keys` and merge their list of custom fields with the default array. This is only fired during the query generation in `get_job_listings()` and I think also the RSS feed.

**Pull request merged**